cli/config/configurator.go

```diff
@@ -187,8 +190,8 @@ func (c Configurator) populateConfigWithDevConfig(ctx context.Context, cfg *Conf
 }
 
 func (c Configurator) populateConfigWithVersionInfo(ctx context.Context, cfg Config) (_ Config, _ error, isOSS bool) {
-	cliVersion := Version
-	if cliVersion == "dev" {
+	useDevVersion := os.Getenv("TRACETEST_AGENT_DEV_CONFIG") == "true"
```

Comment:
it took me a good 20 mins of debuggin to figure out this was happening. So I changed the behavior to be opt in by passing an env var. Otherwise it might hide issues with our config logic during development